lightkurve/lightcurve.py

```diff
@@ -158,15 +158,24 @@ def flatten(self, window_length=101, polyorder=3, return_trend=False, **kwargs):
             Trend in the lightcurve data
         """
         lc_clean = self.remove_nans()  # The SG filter does not allow NaNs
-        trend_signal = signal.savgol_filter(x=lc_clean.flux,
-                                            window_length=window_length,
-                                            polyorder=polyorder, **kwargs)
-        flatten_lc = copy.copy(lc_clean)
-        flatten_lc.flux = lc_clean.flux / trend_signal
-        flatten_lc.flux_err = lc_clean.flux_err / trend_signal
+        #Find breaks in time
+        dt = lc_clean.time[1:]-lc_clean.time[0:-1]
+        cut = np.where(dt > 5*np.nanmedian(dt))[0]+1
```

Comment:
I was thinking of just putting it in the documentation. But you're right, we can expose it and have users tune it. By setting it to a nan I bet you can turn off this functionality........

Reply:
Oh good, let's mention that ``By setting it to a nan I bet you can turn off this functionality`` in the docstrings

Reply:
Ok there's now a break_tolerance keyword which if passed None (or np.nan) will turn off the new behavior. @barentsen this will address your concern about non-uniform time, user just turns it off